bot/converters.py

```diff
@@ -217,7 +217,10 @@ async def convert(self, ctx: Context, duration: str) -> datetime:
         delta = relativedelta(**duration_dict)
         now = datetime.utcnow()
 
-        return now + delta
+        try:
+            return now + delta
+        except ValueError:
+            raise BadArgument(f"`{duration}` results in a datetime outside the supported range.")
```

Comment:
Since there seems to be two ways for the custom `BadArgument` exception to be raised from this method, I would recommend explicitly testing for each of them in `test_converters.py` with `assertRaisesRegex()` to ensure the correct one occurs. There seems to be an existing test for the former in: 

https://github.com/python-discord/bot/blob/6e97a6af9a72286b9ec8102e1abd64f634fe8d92/tests/bot/test_converters.py#L193-L199.

(Note that the above *msg* kwarg in `assertRaises()` does not test the message itself; instead, it customizes the message displayed when the assertion fails. That can be quite useful for diagnosing test failures, but doesn't help much with ensuring the correct exception message occurred.)

Reply:
Good catch, thanks. Didn't realises there were even any tests for converters.